### ISSUE TITLE
Error probing transform on connection

### ISSUE THREAD

**Issue by drasmuss** [nengo/nengo#667]
To replicate:
```python
import nengo

with nengo.Network() as net:
    input = nengo.Node([0])

    a = nengo.Ensemble(10, 1)
    conn = nengo.Connection(input, a)
    
    nengo.Probe(conn, "transform")
    
sim = nengo.Simulator(net)
```
gives error:
```
Traceback (most recent call last):
  File "D:\Documents\officesvn\code\test.py", line 11, in <module>
    sim = nengo.Simulator(net)
  File "D:\Documents\nengogit\nengo\nengo\simulator.py", line 111, in __init__
    self.model.build(network)
  File "D:\Documents\nengogit\nengo\nengo\builder\builder.py", line 34, in build
    return Builder.build(self, *objs)
  File "D:\Documents\nengogit\nengo\nengo\builder\builder.py", line 74, in build
    cls.builders[obj_cls](model, obj, *args, **kwargs)
  File "D:\Documents\nengogit\nengo\nengo\builder\network.py", line 76, in build_network
    model.build(probe)
  File "D:\Documents\nengogit\nengo\nengo\builder\builder.py", line 34, in build
    return Builder.build(self, *objs)
  File "D:\Documents\nengogit\nengo\nengo\builder\builder.py", line 74, in build
    cls.builders[obj_cls](model, obj, *args, **kwargs)
  File "D:\Documents\nengogit\nengo\nengo\builder\probe.py", line 76, in build_probe
    synapse_probe(model, key, probe)
  File "D:\Documents\nengogit\nengo\nengo\builder\probe.py", line 37, in synapse_probe
    sig = sig[probe.slice]
  File "D:\Documents\nengogit\nengo\nengo\builder\signal.py", line 178, in __getitem__
    return self.__getitem__((item,))
  File "D:\Documents\nengogit\nengo\nengo\builder\signal.py", line 151, in __getitem__
    start, stop, stride = idx.indices(shape[ii])
IndexError: list index out of range
```

The problem is that the default transform is a zero dimensional array, so it breaks if you try to slice it (even if it's the default null slice).

**Comment by hunse** [comment]
Zero-dimensional arrays are the bane of my existence :rage1: :rage2: :rage3: :rage4:

**Comment by hunse** [comment]
We could probably just give transforms an `ndmin` of 1 or 2 in the parameter. This might actually simplify the code in some places.

**Comment by tbekolay** [comment]
Fixed in #671.
